aten/src/ATen/cuda/detail/TensorInfo.cuh

```diff
@@ -66,157 +71,70 @@ TensorInfo<T, IndexType>::reduceDim(int dim) {
 
 template <typename T, typename IndexType>
 int
-TensorInfo<T, IndexType>::collapseDims(int excludeDim) {
-  // Find the innermost dimension not of size 1, since dimensions of size 1 are
-  // collapsible.
-  int firstNonOneDim = -1;
-
-  for (int i = dims - 1; i >= 0; --i) {
-    if (i == excludeDim) {
-      // We cannot collapse this dimension, even if it is size 1
-      firstNonOneDim = i;
-      break;
-    }
+TensorInfo<T, IndexType>::collapseDims(const int excludeDim) {
 
-    if (sizes[i] != 1) {
-      firstNonOneDim = i;
-      break;
-    }
-  }
+  int stopDim = (excludeDim == -1) ? dims : excludeDim;
+  int nidx = -1;
```

Comment:
can you make the names a little more descriptive?

maybe `nidx -> newIndex`, `cidx -> oldIndex`, `rVal -> remapDim` say?